apps/zbugs/api/index.ts

```diff
@@ -114,10 +117,25 @@ fastify.post<{
     appID: string;
   };
 }>('/api/push', async function (request, reply) {
-  const response = await pushHandler(
-    {
-      authorization: request.headers.authorization,
-    },
+  let {authorization} = request.headers;
+  if (authorization !== undefined) {
+    assert(
+      authorization.toLowerCase().startsWith('bearer '),
+      'Authorization header must start with `Bearer `. This is a bug in the Zero Pusher service.',
+    );
+    authorization = authorization.substring('Bearer '.length);
+  }
+
+  const jwk = process.env.VITE_PUBLIC_JWK;
+  const authData: AuthData | undefined =
+    jwk && authorization
+      ? authDataSchema.parse(
+          (await jwtVerify(authorization, JSON.parse(jwk))).payload,
+        )
+      : undefined;
```

Comment:
The added boilerplate here seems offset by the gains in the client API. In that they can pass arbitrary data in and don't have to rely on a token.